api/v1alpha1/nodepool_types.go

```diff
@@ -424,3 +426,8 @@ type AgentNodePoolPlatform struct {
 	// +optional
 	AgentLabelSelector *metav1.LabelSelector `json:"agentLabelSelector,omitempty"`
 }
+
+type AzureNodePoolPlatform struct {
```

Comment:
we should make sure we revisit this, but do we need any of the following:

zones (list of az that can be used)
osDisk.diskSizeGB (storage for the instance)